### ISSUE TITLE
Upgrading cluster tools fails with `Failed while: catalog operation fetch`

### ISSUE THREAD

**Issue by abh** [rancher/rancher#37437]
**Rancher Server Setup**
- Rancher version: 2.6.3-patch2
- Installation option (Docker install/Helm Chart):
   - rancher on k3s

**Information about the Cluster**
- Kubernetes version: RKE1, 1.21.9
- Cluster Type (Local/Downstream):
   - Downstream, RKE1 installed, bare metal install

**Describe the bug**

Upgrading rancher monitoring succeeds, but the UI says it failed:

`Failed while: catalog operation fetch`

**To Reproduce**

Cluster Tools -> monitoring -> Upgrade

Different cluster tool, but otherwise this is the same issue as #36331 (which was closed by the stale bot).

**Comment by abh** [comment]
This is still an issue.

**Comment by Zim-b7** [comment]
any update on this? have the same issue with helm charts

**Comment by theAkito** [comment]
I'm getting this sometimes with a custom Helm Chart.

At first, I thought the Helm Chart is just too old or something. Then I modernised it & it worked.

Now, I pushed another version. It does not work anymore & shows this error, except the Chart itself is fine now & worked the last version, even though this version is just a tiny patch version, which changed a couple of values inside the `values.yaml`....

```
Failed waiting for: catalog operation fetch
```

Did anyone manage to solve this, yet?
